stdlib/src/main/scala/chisel3/std/tilelink/TLBundleParameter.scala

```diff
@@ -0,0 +1,23 @@
+package chisel3.std.tilelink
+
+import chisel3.util.isPow2
+
+/**
+  * Parameter of a TileLink link bundle
+  *
+  * All width values are specified in bits
+  *
+  * @param isTLC whether the link is TL-C (cached) level or one of TL-UL, TL-UH (uncached) levels
+  */
+case class TLBundleParameter(
+  addressWidth: Int,
+  sourceWidth:  Int,
+  sinkWidth:    Int,
+  dataWidth:    Int,
+  sizeWidth:    Int,
+  isTLC:        Boolean) {
+  require(dataWidth % 8 == 0, "Width of data field must be multiples of 8")
+  require(isPow2(dataWidth / 8), "Width of data field in bytes must be power of 2")
+
+  def bundle(): TLBundle = new TLBundle(copy())
```

Comment:
I would prefer to not have this in the `case class`. Case classes are supposed to only carry data and be function arguments. I would prefer to keep this simple and not add any methods to the case class.